### ISSUE TITLE
cannot bitcast from enum to packed struct

### ISSUE THREAD

**Issue by nektro** [ziglang/zig#18462]
### Zig Version

0.12.0-dev.1828+225fe6ddb

### Steps to Reproduce and Observed Behavior

```zig
test {
    const q: E = @enumFromInt(242892);
    const r: P = @bitCast(q);
    _ = r;
}

const E = enum(u32) {
    _,
};

const P = packed struct(u32) {
    a: u16,
    b: u16,
};
```

```
❯ zig test test.zig 
test.zig:23:27: error: cannot @bitCast from 'test.E'
    const r: P = @bitCast(q);
                          ^
```

### Expected Behavior

success

**Comment by nektro** [comment]
errata: the other direction doesn't work either but its far easier to workaround with `@bitCast(@intFromEnum(e))` since the intFromEnum call doesn't need a result type

**Comment by Vexu** [comment]
I would expect this to be an intentional extension of #3647

**Comment by nektro** [comment]
does that mean `@enumFromInt` should/does accept packed structs directly? I'll have to try that

---

: ( alas no, I get `error: expected integer type, found 'P'`

**Comment by Vexu** [comment]
I don't think so, packed structs are not ints despite being implemented as such.

**Comment by nektro** [comment]
I think if https://github.com/ziglang/zig/issues/18882#issuecomment-1936961406 is implemented then this starts to make a lot of sense
